### PR TITLE
Qcon one

### PR DESCRIPTION
**Pull request by kevinhodges**
### Blog title
QCon New York 2015 key takeaways - Part 1

### Blog description
First part (of 2) summarising my notes from this years QCon in New York.

#### Author
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 1
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 2 @notABluesSinger
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

### CODE REVIEW

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+layout: post
+title: QConn New York 2015 key takeaways - Part 1
```
Comment (Joezo) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: I think `Qconn` should be `QCon`

**Comment by JonathanBD** [comment]
I'll be Editor #1

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+---
+
+#### Introduction
+
+This is the first part of my notes and key take aways from Qonn New York 2015, a three day conference with main features (for me) on microservices, containers and culture.
+
+### Automating operational decisions in real-time @ Netflix
+#### Chris Sanden
+
+Kev: Really interesting talk to see how "the big boys" keep track of things (services/services/customers) at scale.  Feels like we're a long way from this, we have visibility over what we're doing but would take a fair amount of work, understanding, patience and maintenance to put something that is useful in to production.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: What does the "Kev:" bit mean? All the entry appears to be from Kev's POV, so can probably drop this.

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+title: QCon New York 2015 key takeaways - Part 1
+categories: conference
+author: kevin_hodges
+tags: conference qcon new york 2015
+comments: true
+---
+
+#### Introduction
+
+This is the first part of my notes and key take aways from Qonn New York 2015, a three day conference with main features (for me) on microservices, containers and culture.
```
Comment (JonathanBD) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: three-day

File: _posts/2015-07-14-qconn-new-york-part-1.md (hunk 2 above)
Comment (JonathanBD) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: "Feels like we're a long way from this; we have....."

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+- Vicious cycle of viewing, learning and improving the experience.
+- Too many stateful instances in constant use, wanted better scaling and less waste.
+
+Principles for more to new system
+
+- Architectures are "throwawayable" or "limited time" and based on a collect, process, provider pattern.
+- Continuity is paramount
+
+How?
+- Small iterations
```
Comment (JonathanBD) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Full stops after bullets. Works with/without them, but inconsistent across post.

**Comment by JonathanBD** [comment]
All done. Edits suggested.

File: _posts/2015-07-14-qconn-new-york-part-1.md (hunk 3 above)
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Qonn

File: _posts/2015-07-14-qconn-new-york-part-1.md (hunk 2 above)
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: services twice... did you mean services/servers/customers?

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+### Automating operational decisions in real-time @ Netflix
+#### Chris Sanden
+
+Kev: Really interesting talk to see how "the big boys" keep track of things (services/services/customers) at scale.  Feels like we're a long way from this, we have visibility over what we're doing but would take a fair amount of work, understanding, patience and maintenance to put something that is useful in to production.
+
+Netflix deal with big numbers when it comes to metrics, not just throughput but also what "keeping things up" looks like.  In terms of humans this simply doesn't scale.  Decisions need to be repetable and incremental, not "squishy".  
+
+Also, metrics naturally "drift" over time due to things like the internet, the market, load.  Detecting tiny incremental changes manually is impossible but machines can "learn" what the boundaries of normal are and tweak themselves based on existing use cases.  Netflix feed "tagged" data to their algorithms that "know" what they are looking for and can adjust themselves automatically to help keep servers up or detect anomalies.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Double space before `Detecting`

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+Example, when an EC2 instance boots it may simply be faulty, this needs to be detected in real-time and destroyed before it's put in front of customers.  Each time a server is tested a "score" is generated by the automated tests and if it's <95% to the control server it will be binned.
+
+### Netflix’s Viewing Data Microservices: How We Know Where You Are in House of Cards
+#### Matt Zimmer
+
+Kev: Great talk and speaker, really interesting guy. I definitley feel this is something we should be thinking about a lot more in our business, even if we don't implement it fully or in a stripped down/different way, we should be thinking about the systems we control in a different way.  Systems shouldn't represent/dictate implementation, microservices shouldn't work like that.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Bit of repetition with 'different way' here

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Once happy, the stateful tier was then scaled down
+
+GEN5 hunch, ever smaller systems with on iteration on the collect, process, provider theory.
+
+
+### Microservices and the art of taming the Dependency Hell Monsters
+#### Michael Bryzek
+
+Kev: We're doing a lot of this stuff in parts of the business, we need to be sure this continues to proliferate and permeate all facets of the web team.  It's "general practice" in open source for  reason.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: missing an 'a' at 'open source for  reason'

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+Works for Gilt, "American ASOS" with a big dev culture similar to Etsy.
+
+https://dev.gilt.com/
+
+Internally they have many small RPC style services, product-catalog, inventory-service, cart-service, users.
+Problem is, as they become mature, the builds get larger and slower, each service has a different "play book", increased boilerplate, quality drops which slows down development
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Full-stop at sentence end is missing

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+- backwards compatibility
+- minimal external libraries
+- know when things change
+- consistent naming (REST)
+- smaller/flatter services so less dependencies
+
+### Managing Complexity - Functionality
+#### Ryan Trinkle
+
+Kev: Best speaker of the conference, sold his idea well and will be looking in to functional programming more to see if we can adapt and patterns to what we're doing with JS.
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: and patterns or any patterns?

File: _posts/2015-07-14-qconn-new-york-part-1.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+Move fast and break things is fine for some but in practice, we want to move fast  without breaking.  And software is complex so this is inherently difficult in two major ways:
```
Comment (notABluesSinger) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: double space in 'move fast without breaking'

File: _posts/2015-07-14-qconn-new-york-part-1.md (hunk 3 above)
Comment (kevinhodges) [inline _posts/2015-07-14-qconn-new-york-part-1.md]: Qcon is single n :)

**Comment by kevinhodges** [comment]
All comments actioned or fed back on where necessary, thanks for the tips guys :)

**Comment by JonathanBD** [comment]
Looks good

**Comment by notABluesSinger** [comment]
:+1: from me

**Pull request merged**